src/emqx_passwd.erl

```diff
@@ -14,12 +14,36 @@
 
 -module(emqx_passwd).
 
--export([hash/2]).
+-export([hash/2, check_pass/2]).
 
 -type(hash_type() :: plain | md5 | sha | sha256 | pbkdf2 | bcrypt).
 
 -export_type([hash_type/0]).
 
+-spec(check_pass(binary() | tuple(), binary() | tuple()) -> binary()).
+check_pass({PassHash, Password}, bcrypt) ->
+	case binary:part(PassHash, {0,29}) of		
+		{error, Error}->
+            error_logger:error_msg("bcrypt hash error:~p", [Error]),
+            <<>>;
+		Salt ->
+			check_pass(PassHash, emqx_passwd:hash(bcrypt, {Salt, Password}))
+    end;
```

Comment:
align the end with case

Reply:
Could you please add test cases for new functions?

Reply:
The test cases are in the auth plugins like emqx_auth_mysql. I have another PR for emqx_auth_mysql, there is a test case for bcrypt hashtype.